node/pkg/near/watcher.go

```diff
@@ -0,0 +1,381 @@
+				if err != nil {
+					return err
+				}
+
+				observation := &common.MessagePublication{
+					TxHash:           txHash,
+					Timestamp:        time.Unix(int64(ts), 0),
+					Nonce:            uint32(event_json.Get("nonce").Uint()), // uint32
+					Sequence:         event_json.Get("seq").Uint(),
+					EmitterChain:     vaa.ChainIDNear,
+					EmitterAddress:   a,
+					Payload:          pl,
+					ConsistencyLevel: 0,
+				}
+
+				nearMessagesConfirmed.Inc()
+
+				logger.Info("message observed",
+					zap.Uint64("ts", ts),
+					zap.Time("timestamp", observation.Timestamp),
+					zap.Uint32("nonce", observation.Nonce),
+					zap.Uint64("sequence", observation.Sequence),
+					zap.Stringer("emitter_chain", observation.EmitterChain),
+					zap.Stringer("emitter_address", observation.EmitterAddress),
+					zap.Binary("payload", observation.Payload),
+					zap.Uint8("consistency_level", observation.ConsistencyLevel),
+				)
+
+				e.msgChan <- observation
+			}
+		}
+	}
+
+	return nil
+}
+
+func (e *Watcher) inspectBody(logger *zap.Logger, block uint64, body gjson.Result) error {
+	logger.Info("inspectBody", zap.Uint64("block", block))
+
+	result := body.Get("result.chunks.#.chunk_hash")
+	if !result.Exists() {
+		return nil
+	}
+
+	v := body.Get("result.header.timestamp")
+	if !v.Exists() {
+		return nil
+	}
+	ts := uint64(v.Uint()) / 1000000000
+
+	for _, name := range result.Array() {
+		chunk, err := e.getChunk(name.String())
+		if err != nil {
+			return err
+		}
+
+		txns := gjson.ParseBytes(chunk).Get("result.transactions")
+		if !txns.Exists() {
+			continue
+		}
+		for _, r := range txns.Array() {
+			hash := r.Get("hash")
+			receiver_id := r.Get("receiver_id")
+			if !hash.Exists() || !receiver_id.Exists() {
+				continue
+			}
+
+			err = e.inspectStatus(logger, hash.String(), receiver_id.String(), ts)
+			if err != nil {
+				return err
+			}
+		}
+
+	}
+	return nil
+}
+
+func (e *Watcher) Run(ctx context.Context) error {
+	p2p.DefaultRegistry.SetNetworkStats(vaa.ChainIDNear, &gossipv1.Heartbeat_Network{
+		ContractAddress: e.wormholeContract,
+	})
+
+	logger := supervisor.Logger(ctx)
+	errC := make(chan error)
+
+	logger.Info("Near watcher connecting to RPC node ", zap.String("url", e.nearRPC))
+
+	go func() {
+		if e.next_round == 0 {
+			finalBody, err := e.getFinalBlock()
+			if err != nil {
+				logger.Error("StatusAfterBlock", zap.Error(err))
+				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDNear, 1)
+				errC <- err
+				return
+			}
+			e.next_round = gjson.ParseBytes(finalBody).Get("result.chunks.0.height_created").Uint()
+		}
+
+		timer := time.NewTicker(time.Second * 1)
+		defer timer.Stop()
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case r := <-e.obsvReqC:
+				if vaa.ChainID(r.ChainId) != vaa.ChainIDNear {
+					panic("invalid chain ID")
+				}
+
+				txHash := base58.Encode(r.TxHash)
+
+				logger.Info("Received obsv request", zap.String("tx_hash", txHash))
```

Comment:
Can you add the chain name to these logs please? It makes it much harder for some of these more generic messages to troubleshoot or just see what's happening.

Reply:
isn't the filename in the logs?  Is there a standard on how you want to see the chain?  I don't see the chain in the "message observed" message for solana either

Reply:
Yeah, this is one of those things we really badly need to just burn through and refactor. We can do it in a followup.